### DISCUSSION TITLE
NoSuchKey - 404 errors when fetching images from MinIO

### DISCUSSION THREAD

**Discussion by Limezy** [outline/outline#4918] [Self Hosting] [answered]
Hi all, 
After migrating my Debian to Debian 11, my outline instance which was working perfectly, now isn't able to fech images anymore.
I get a 404 error with NoSuchKey - The specified key does not exist when I charge a page of the wiki :

![322551441d0bbd0a93451c2cb22372637fa94196](https://user-images.githubusercontent.com/24638389/220963552-08e87e80-53d2-4bad-a2b4-bf5a9d889725.jpeg)

The strange part being that everything works perfectly for *new* images or attachments, that I upload now. Only the "old" ones, that I had uploaded before, do have that problem.

I have checked, and I confirm that I can see the JPG image when going to the storage of the MinIO bucket, and that the image has the right UNIX permissions. Any hint or idea on when this problem could come from ?

Thanks in advance for your help !

**Comment by tommoor** [comment, marked as answer]
This feels more like an issue with Minio than Outline, I'd try generating some signed urls from the minio admin and compare how they differ perhaps?

**Reply by Limezy** [reply]
Many thanks for your answer. You were right, this was caused by a failed migration of MinIO, from the older backend model to the new "SNSD" one (by the way, that migration is a pain in the a** for casual users 🤣)
